cfg/diag-x16.cfgtpl

```diff
@@ -0,0 +1,12 @@
+MEMORY {
+	#include "x16.cfginc"
+
+	DIAG:	start=$C000, size=$4000, fill=yes, fillval=$AA;
+}
+
+SEGMENTS {
+	DIAG:		load= DIAG, type=ro;
+	/* address of start label in init.s, found in kernal.sym */
+	ROMINIT:	load=DIAG, type=ro, start=$E997;
```

Comment:
This appears to be brittle.  If we're going to trampoline like this in ROM, I'd much rather have it at a well-defined address that is unlikely to shift.

I would suggest changing `kernal-x16.cfgtpl` and moving INIT to the first line after SIGNATURE, so it is the first segment to load KERNAL.  Add an assert to `kernal/cbm/init.s` at the `start` label, such as

`.assert * = $C010, error, "kernal init must start at $C010"`

similarly, put your diag entry point at $C010 as well, perhaps with a similar `.assert`.

Reply:
I need more guidance on this one... Maybe you could help me in discord?